### PR TITLE
Refactor Sqlalchemy queries to 2.0 style (Part 3)

### PR DESCRIPTION
**Pull request by phanikumv**
This is a continuation of the effort to refactor the queries to sqlalchemy 2.0 (Re-applying the revert done in https://github.com/apache/airflow/pull/32343 by applying the label `full tests needed`)

cc @ephraimbuddy @uranusjr @potiuk 

related: https://github.com/apache/airflow/issues/28723

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Same MSSQL error as we saw in main earlier today

**Review by ephraimbuddy — APPROVED** [review]

**Review by uranusjr — APPROVED** [review]

**Comment by phanikumv** [comment]
@potiuk can we merge this now

**Comment by potiuk** [comment]
Oh absolutely :)

File: airflow/www/views.py [hunk 1]
```diff
@@ -2293,15 +2300,17 @@ def clear(self, *, session: Session = NEW_SESSION):
...
+            dag_runs_query = session.scalars(
+                select(DagRun.id).where(DagRun.dag_id == dag_id).with_for_update()
+            )
             if start_date is None and end_date is None:
-                dag_runs_query = dag_runs_query.filter(DagRun.execution_date == start_date)
+                dag_runs_query = dag_runs_query.where(DagRun.execution_date == start_date)
             else:
                 if start_date is not None:
-                    dag_runs_query = dag_runs_query.filter(DagRun.execution_date >= start_date)
+                    dag_runs_query = dag_runs_query.where(DagRun.execution_date >= start_date)
```
Comment (bbovenzi) [inline airflow/www/views.py:2310]: I'm not too familiar with sqlalchemy 2.0 but this line is throwing an error and is breaking the ability to clear task groups in the UI
`AttributeError: 'ScalarResult' object has no attribute 'where'`

Comment (phanikumv) [reply airflow/www/views.py:2310]: Fixed in #32801

**Pull request merged**